### ISSUE TITLE
Action Groups Extended integration documentation 

### ISSUE THREAD

**Issue by erendrake** [KSP-KOS/KOS#487]
Now that we have an awesome new documentation page and we are integrated with two other mods. I think we should have some documentation for that integration. 

This should cover:

* a clear statement about why we integrated
* the current features with some screenshots/video
* compatibility info
* future plans for the integration

**Comment by SirDiazo** [comment]
And here is what should be functionally the final version.

https://github.com/SirDiazo/KOS/blob/develop/doc/source/addons/AGX.rst

Probably needs and editing and formatting pass and the external links are not all there yet, but the page essentials are there and I don't forsee anything major changing at this point.

However, does it line up with what you expect for docs and how do I do newLines and code blocks?

**Comment by Dunbaratu** [comment]
I was largely not paying attention to this because I've had my head stuck in telnet for a few days, but I just thought of something:

When I added the suffixes :PARTSINGROUP and :MODULESINGROUP to VesselTarget, At the time they were hardcoded to only know about the action groups in stock, and had a bit of ugly cut-n-paste code shared between them, with a comment about refactoring that code later.

Can we make :PARTSINGROUP and :MODULESINGROUP work with AG11 through AG250?

**Comment by SirDiazo** [comment]
Easily.

Is a List<Part> and List<PartModule> okay? And do you want duplicates if their are multiple actions on the same part, or only return a specific Part/PartModule once?

**Comment by erendrake** [comment]
@Dunbaratu thanks for bringing this up. I was thinking about releasing setting and getting the state of these action groups and then coming back to AGX and add support for PIG and PIM in a subsequent release. Talking about it now is an excellent idea.

@SirDiazo 

    Is a List[Part] and List[PartModule] okay?

I think so, we are not exposing the exact action that they had hooked up to that action group. It is simply another strategy for allowing users to query a custom list of parts/modules

     And do you want duplicates if their are multiple actions on the same part, or only return a specific Part/PartModule once?

for our uses i would de-dupe the lists before sending them to the users. I will let you decide what makes the most sense for AGX. 

    Actually, what do you want to see back, all the Parts/PartModules in a specified action group, or a list of a Part/PartModules with an assigned action and the group they are in?

Right now because we dont show the 'actions' (eg leg up, deploy chute, deactivate engine)  that are bound we simply return the parts/modules. so the former is what we need. the later would be more useful for action group discovery or display. which we dont really have on our roadmap.

**Comment by erendrake** [comment]
@SirDiazo The two quick start images are hosted externally to the documentation. I am fine with external links like the forums and such. Would you mind putting images into the documentation project?

**Comment by SirDiazo** [comment]
Alright, tried to add the image link. Is there anyway I can test it without having to merge it to the official docs?

**Comment by Dunbaratu** [comment]
The intent of :PARTSINGROUP and :MODULESTAGGED is to create distinct unique lists of parts - no duplications - that is correct.

I admit I haven't looked at the architecture of the external module support system of kOS yet, so I don't quite know how to add the code to the suffixes in a way that runs  fine when AGExt is missing, or where it should get added.

If you want to have a look, the relevant two methods inside kOS are here:

https://github.com/KSP-KOS/KOS/blob/develop/src/kOS/Suffixed/VesselTarget.cs, ```GetPartsInGroup``` on line 296 and ```GeModulesInGroup``` on line 343,
(Line numbers subject to change if anyone checks in a change after I write this, of course).

One interesting question could be... since AGExt provides two different ways to name an action group - do we want to allow the query to use both techniques so either name works?  I'd say yes.  That would mean that SHIP:PARTSINGROUP("AG15") returns the same thing as SHIP:PARTSINGROUP("The custom Name I gave to Action group 15").


Setting up a local copy of the docs and regenerating them is a nontrivial bit of setup work that only exists in emails exchanged between some of us (by the way, that needs to change- the docs about the docs need to be added to the docs), so I think as long as we can guess what you meant it to look like, we can probably clean it up just fine, and be happy you're willing to write docs at all

In fact we could easily enough download the image file and add it ourselves and re-do the link, except for legal reasons.  Adding an image we downloaded from a URL, that mentions no licensing terms, doesn't *quite* give us proper permission to republish it..  If you add it to the repository yourself, and the repository is stated as being covered under GPL3, then that constitutes (I assume, I'm no lawyer) sufficient evidence that you are willingly putting your images under GPL3 terms and we didn't "steal" them.
